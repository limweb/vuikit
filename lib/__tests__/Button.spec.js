import { destroyVM, createVue, toJSON } from './_util'

describe('Button', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    vm = createVue(`
      <vk-button type="submit" color="primary" width="1-1" size="mini" :active="true" :disabled="true">
        Button
      </vk-button>
    `)
    expect(toJSON(vm)).toMatchSnapshot()
  })

  it('triggers event click when the button was clicked', () => {
    const vm = createVue(`<vk-button />`)
    const button = vm.$children[0]
    const cb = jest.fn()
    button.$on('click', cb)
    button.$el.click()
    expect(cb).toHaveBeenCalled()
    expect(cb.mock.calls[0].length).toBe(1)
  })
})
