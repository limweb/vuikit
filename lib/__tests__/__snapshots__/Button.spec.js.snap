exports[`Button renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": Array [
          "uk-button",
          "uk-active",
          "uk-button-primary",
          "uk-button-mini",
          "uk-width-1-1",
        ],
        "disabled": "disabled",
        "type": "submit",
      },
      "child": Array [
        Object {
          "node": "text",
          "text": "Button",
        },
      ],
      "node": "element",
      "tag": "button",
    },
  ],
  "node": "root",
}
`;
